src/nrniv/nrncore_write/data/cell_group.cpp

```diff
@@ -110,7 +116,19 @@ CellGroup* CellGroup::mk_cellgroups(CellGroup* cgs) {
                     } else { // if an acell is never a source, it will not have a presyn
                         cgs[i].output_gid[npre] = -1;
                     }
-                    // the way we associate an acell PreSyn with the Point_process.
+                    // the way we associate an acell PreSyn with the
+                    // Point_process.
+                    if (agid < std::numeric_limits<int>::min() ||
+                        agid > std::numeric_limits<int>::max()) {
```

Comment:
Seems like this test allows positive agid. I think it should be agid > = -1

Reply:
OK, I pushed that ("integer overflow" might be a little misleading in that case, but I think it's good enough).